docs/topics/Gateway.md

```diff
@@ -561,27 +561,33 @@ When connecting to the gateway as a bot user, guilds that the bot is a part of w
 
 ## Sharding
 
-As apps grow and are added to an increasing number of guilds, some developers may find it necessary to divide portions of their app's operations across multiple processes. As such, the Gateway implements a method of user-controlled guild sharding which allows apps to split events across a number of Gateway connections. Guild sharding is entirely controlled by an app, and requires no state-sharing between separate connections to operate. While all apps *can* enable sharding, it's not necessary for apps in a smaller number of guilds. 
+As apps grow and are added to an increasing number of guilds, some developers may find it necessary to divide portions of their app's operations across multiple processes. As such, the Gateway implements a method of user-controlled guild sharding which allows apps to split events across a number of Gateway sessions. Guild sharding is entirely controlled by an app, and requires no state-sharing between separate sessions to operate. While all apps *can* enable sharding, it's not necessary for apps in a smaller number of guilds.
 
 > warn
-> Each shard can only support a maximum of 2500 guilds, and apps that are in 2500+ guilds *must* enable sharding. 
+> Each shard can only support a maximum of 2500 guilds, and apps that are in 2500+ guilds *must* enable sharding.
 
-To enable sharding on a connection, the app should send the `shard` array in the [Identify](#DOCS_TOPICS_GATEWAY_EVENTS/identify) payload. The first item in this array should be the zero-based integer value of the current shard, while the second represents the total number of shards. DMs will only be sent to shard 0.
+Sessions that would like to only receive events from a subset of guilds should send the `shard` array in the [Identify](#DOCS_TOPICS_GATEWAY_EVENTS/identify) payload. The first item in this array is `shard_id`, the zero-based integer value of the current shard, while the second is `num_shards` and represents the total number of shards.
 
 > info
 > The [Get Gateway Bot](#DOCS_TOPICS_GATEWAY/get-gateway-bot) endpoint provides a recommended number of shards for your app in the `shards` field
 
-To calculate which events will be sent to which shard, the following formula can be used:
+A certain gateway session is only subscribed to events from guilds with a `guild_id` that fulfil the following formula, using the `shard_id` and `num_shards` that the session provided in the [Identify](#DOCS_TOPICS_GATEWAY_EVENTS/identify) event:
 
 ###### Sharding Formula
 
 ```python
-shard_id = (guild_id >> 22) % num_shards
+shard_id == (guild_id >> 22) % num_shards
```

Comment:
```suggestion
shard_id = (guild_id >> 22) % num_shards
```

that change is incorrect. `==` is an equality operator, `=` sets the `shard_id` SO it was correct.

Reply:
https://github.com/Pycord-Development/pycord/blob/08370b57368e2e5ce0e4e70faf8514742f45f5e5/discord/guild.py#L1011-L1016

https://github.com/DSharpPlus/DSharpPlus/blob/cc238ce9542ab1ef2137dd4816a6062cb4ea933c/DSharpPlus/Utilities.cs#L196-L197

https://github.com/discordjs/discord.js/blob/e673b3c129f288f9f271e0b991d16dc2901cdc8a/packages/util/src/functions/calculateShardId.ts#L7-L9

Reply:
The most common usage is imo calculating based on the guild id, on which shard the guild is located on, hence I suggest not changing it

Reply:
> This example is meant to calculate the shard id for given guild_id, not to check the equality.

No, the whole section preceding this formula explains which events will be received by a specific shard *(and not which shards will receive a specific event)*. So you should look at the formula in the context of a shard (the shard_id is known in this case). The previous version of this section was causing confusion to devs that aren't proficient with how Discord is working; you should [take a look at the original question](https://discord.com/channels/613425648685547541/1208012213491736576/1208012213491736576) that ended up in the creation of this PR.

Anyway, your arguments about how lib use this formula don't make sense: the point of the documentation, especially for something that may be complex to understand, is to explain how this works, not to have code and formulas people can just copy&paste without thinking. However, I agree this may interpreted as a typo, so to make it clear this is an equation, I suggest to do this:
```suggestion
(guild_id >> 22) % num_shards == shard_id
```

Reply:
that sounds like a good way

Reply:
345

Reply:
That seems reasonable. Thank you.